### PR TITLE
Create major.minor tag for the latest major.minor.x.x mandrel version

### LINKED ISSUE #91: Create major.minor (e.g. 20.1-java11) floating tag in quay for mandrel images
As discussed in https://github.com/graalvm/mandrel/issues/95 we would like to have "floating" tags for mandrel images.

The floating tags would be based on the `major.minor` part of the tag and point to the latest `major.minor.micro.patch.Final` tag, i.e., `20.1-java11` would point to `20.1.0.1.Final-java11` today and later to `20.1.0.2.Final` when released.

Ideally this process should be automated in https://github.com/quarkusio/quarkus-images/blob/master/.github/build-mandrel-images.sh but I see the following issue.
When a new image is created and pushed on quay it is not made public until someone does this manually. 
If we are to automate the update of the floating tags how would that work? 
We would essentially have the floating tag (which should be public) point to a private tag and thus break builds until the new tag becomes public.

\cc @galderz @cescoffier

### PR DESCRIPTION
**Pull request by zakkak**
Closes #91 and https://github.com/graalvm/mandrel/issues/95

### CODE REVIEW

File: .github/build-mandrel-images.sh [hunk 1]
```diff
@@ -17,5 +17,12 @@ do
         ${BUILD_ENGINE} --tag="${PREFIX_NAME}:${version}"
 done
 
+# Get all the final versions, reverse sort them, and keep only the first for each major.minor (e.g. 20.1)
+LATEST_FINAL_VERSIONS=$(sed 's/ /\n/g' <<< ${VERSIONS[@]} | grep Final | sort -r | sort -k1,1 -k2,2 -k5,5 -t'.' --unique)
```
Comment (cescoffier) [inline .github/build-mandrel-images.sh:21]: Just "wow" ... :-D

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Not related to this PR, but I was thinking of replacing the bash script with jbang. We start doing more fancy stuff in the script and bash is not necessarily easily maintainable. 

What's your opinion on this?

**Comment by cescoffier** [comment]
Hum, the sed didn't work for me. Looking...

**Comment by zakkak** [comment]
> Not related to this PR, but I was thinking of replacing the bash script with jbang. We start doing more fancy stuff in the script and bash is not necessarily easily maintainable.
> 
> What's your opinion on this?

`LATEST_FINAL_VERSIONS=$(sed 's/ /\n/g' <<< ${VERSIONS[@]} | grep Final | sort -r | sort -k1,1 -k2,2 -k5,5 -t'.' --unique)` is definitely not readable/maintainable. 
Additionally, with `bash` I am always scared that I will use something that won't work on every system (e.g. the `sed` command didn't work for you).
IMHO however I would prefer pure Java scripts, python or some other alternative that is common and does not require setting up `jbang` in the CI.

File: .github/build-mandrel-images.sh (hunk 1 above)
Comment (zakkak) [reply .github/build-mandrel-images.sh:21]: Let's hope the next one who will want to alter/understand it will have the same reaction :laughing:

**Comment by cescoffier** [comment]
There are a way to run jbang scripts directly from github actions. @maxandersen can tell us more about it.

I'm not a fan of python, so I would avoid it.

**Comment by maxandersen** [comment]
https://GitHub.com/jbangdev/jbang-action

Use as action or run via docker.

**Comment by zakkak** [comment]
Nice, then jbang is fine I guess :)

You are doing really great work with jbang @maxandersen . Thanks!

**Comment by maxandersen** [comment]
Thanks :) And good timing as I just last night released update that fixed a bug that made it hard to run in an action. So :latest or :v0.38.0 recommended :)

**Pull request merged**